javascript/javascript1/week2/exerciseStep3Week2.js

```diff
@@ -0,0 +1,138 @@
+//Flight Booking Full Name Function//////////////////////////////////////////////////////////////
+
+function getFullName(firstname, surname, useFormalName) {
+    if (firstname === '') {
+        return 'Please write your first name'
+    } else if (surname === '') {
+        return 'Please write your surname'
+    } else if (useFormalName) {
+        return 'Lord ' + firstname + ' ' + surname;
+    } else {
+        return firstname + ' ' + surname;
+    }
+}
+
+const fullName1 = getFullName('Hamed', 'Karachi', true);
+const fullName2 = getFullName('Arman', 'Naderi', false);
+
+console.log(fullName1);
+console.log(fullName2);
+
+//I do not understand the last section of this exercise when it said we should try out to put it in codepen
+
+
+//Event Application/////////////////////////////////////////////////////////////////////////
+//"i" is the parameter for how many day we have until to the event
+
+const n = new Date().getDay();
+const weekday = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
+
+function getEventWeekday(i) {
+    if (i >= 7) {
+        return ((i % 7) + n) - 1;
+    } else {
+        return ((i + n) % 7) - 1;
+    }
+}
+
+
+//Weather wear//////////////////////////////////////////////////////////////////
+
+function weatherTemprature(temp) {
+    if (temp <= 0) {
+        return 'wear a jacket and gloves with anti sliping shoes'
+    } else if (temp <= 10) {
+        return 'wear a jaacket with trousers'
+    } else if (temp <= 20) {
+        return 'wear a T-shirt with trousers'
+    } else {
+        return 'wear shorts and T-shirt'
+    }
+}
+
+//Student Manager/////////////////////////////////////////////////////////////////
+
+var class07Students = [];
+
+function addStudentToClass(studentName) {
+    if (studentName !== '') {
```

Comment:
This works perfectly, but if you want to take your coding skills to the next level, might I introduce you to [guard clauses](https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/#:~:text=The%20guard%20clause,are%20met%2C%20or%20not%20met.&text=Guard%20clauses%20can%20reduce%20the,certain%20code%20to%20be%20executed.) Basically it is a way of doing away with so many levels of nested `if` statements and makes your code much simpler to read. Check them out and if it not clear, let's talk about them in the next class.